lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -456,7 +456,9 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
                 .variant_names
                 .insert(Symbol::Nonterminal(nt.clone()), name.clone());
         }
-
+        if self.grammar.intern_token.is_some() {
```

Comment:
I think this new case deserves a comment describing why we need that (potentially linking to the issue as well). Other steps have a short one describing what happening, and I imagine it's not entirely clear out of the context of the original issue, why we need to do that in the first place.